### PR TITLE
Bump go version to 1.7.1, since 1.6.3 cannot build working binaries on OS X Sierra

### PR DESCRIPTION
**Pull request by cyli**
:|

See https://groups.google.com/forum/#!msg/golang-dev/Jho5sBHZgAg/cq6d97S1AwAJ

### CODE REVIEW

File: NOTARY_VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.4.1
+0.4.2
```
Comment (riyazdf) [inline NOTARY_VERSION]: Can we also make a quick changelog note?

Comment (cyli) [inline NOTARY_VERSION]: Good point, thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## [v0.4.2](https://github.com/docker/notary/releases/tag/v0.4.0) 9/30/2016
++ Bump the cross compiler to golang 1.7.1, since [1.6.3 builds binaries that could have non-deterministic bugs in OS X Sierra](https://groups.google.com/forum/#!msg/golang-dev/Jho5sBHZgAg/cq6d97S1AwAJ) [#984](https://github.com/docker/notary/pull/984)
```
Comment (riyazdf) [inline CHANGELOG.md]: 👍

**Comment by cyli** [comment]
@justincormack approved.  ping @endophage for one more +1?

**Pull request merged**